### PR TITLE
ODPM-549: kafka - consume partition/offset implementation

### PR DESCRIPTION
**Pull request by VladoLavor**

### CODE REVIEW

File: examples/kafka_plugin/clustered/main.go [hunk 1]
```diff
@@ -115,33 +117,45 @@ type Deps struct {
...
 	// Init channels required for async handler
-	plugin.asyncMessageChannel = make(chan messaging.ProtoMessage, 0)
+	plugin.asyncSuccessChannel = make(chan messaging.ProtoMessage, 0)
 	plugin.asyncErrorChannel = make(chan messaging.ProtoMessageErr, 0)
 
 	connection := plugin.Kafka.NewProtoConnection("example-proto-connection")
 
 	// Create a synchronous publisher for the selected topic and partition
-	plugin.kafkaSyncPublisher = connection.NewSyncPublisherToPartition(topic, 0)
+	plugin.kafkaSyncPublisher = connection.NewSyncPublisherToPartition(topicSync, 0)
```
Comment (jozef-slezak) [inline examples/kafka_plugin/clustered/main.go]: I am thinking, topicSync might be confusing (sync sync is allready used mainly in resync and/or datasync).

File: examples/kafka_plugin/clustered/main.go [hunk 2]
```diff
@@ -112,28 +115,73 @@ type Deps struct {
...
+	plugin.asyncSuccessChannel = make(chan messaging.ProtoMessage, 0)
+	plugin.asyncErrorChannel = make(chan messaging.ProtoMessageErr, 0)
+	plugin.kafkaAsyncPublisher, err = connection.NewAsyncPublisherToPartition(topic2, asyncMessagePartition,
+		messaging.ToProtoMsgChan(plugin.asyncSuccessChannel), messaging.ToProtoMsgErrChan(plugin.asyncErrorChannel))
+	if err != nil {
+		return err
+	}
 
-	// kafkaWatcher.Watch is called to start consuming a topic.
+	// Initialize sync watcher
```
Comment (lukasmacko) [inline examples/kafka_plugin/clustered/main.go]: remove the sync wather. Watcher is always async

File: messaging/kafka/mux/bytes_connection.go [hunk 3]
```diff
@@ -47,17 +47,83 @@ func (conn *Connection) ConsumeTopic(msgClb func(message *client.ConsumerMessage
...
 
 	for _, topic := range topics {
-		// check if we have already consumed the topic and partition
-		subs, found := conn.multiplexer.mapping[topic]
+		// check if we have already consumed the topic
+		var found bool
+		var subs *consumerSubscription
+	LoopSubs:
+		for _, subscription := range conn.multiplexer.mapping {
+			if subscription.clustered == true {
```
Comment (lukasmacko) [inline messaging/kafka/mux/bytes_connection.go]: better name would be probably subscription.manual

File: examples/kafka_lib/mux/main.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ import (
 
 func main() {
 	log.StandardLogger().SetLevel(logging.DebugLevel)
-	mx, err := mux.InitMultiplexer("", "default", logroot.StandardLogger())
+	mx, err := mux.InitMultiplexer("", "default", "", logroot.StandardLogger())
```
Comment (jozef-slezak) [inline examples/kafka_lib/mux/main.go]: please fill:
- name of the connection
- hash (from enum representing hash/manual/random)

File: examples/kafka_plugin/clustered/main.go [hunk 5]
```diff
@@ -112,28 +115,73 @@ type Deps struct {
...
-	// Init channels required for async handler
-	plugin.asyncMessageChannel = make(chan messaging.ProtoMessage, 0)
-	plugin.asyncErrorChannel = make(chan messaging.ProtoMessageErr, 0)
-
-	connection := plugin.Kafka.NewProtoConnection("example-proto-connection")
-
-	// Create a synchronous publisher for the selected topic and partition
-	plugin.kafkaSyncPublisher = connection.NewSyncPublisherToPartition(topic, 0)
+	// Prepare two topics for sync and async scenario
+	topic1 := "example-sync-clustered-topic"
```
Comment (jozef-slezak) [inline examples/kafka_plugin/clustered/main.go]: const...

**Pull request merged**